.github/workflows/trigger-private-janus-build.yml

```diff
@@ -0,0 +1,90 @@
+# This triggers a build in a separate github action
+
+## See documentation here: https://docs.google.com/document/d/1Xz-gzptICIza0oio0Kza3LWg2a6n_015o0MPu0LGfuU/edit#heading=h.k45jb8zdijoi
+## (Only available for select Guardian Staff)
+## This also relies on a personal access token, due to expire on or before 15th June 2024 ( 2024-06-15 ).
+## Details of how to renew this token in the docs.
+
+### This task is broken into 2 jobs
+### in case for whatever reason the internal build takes a long time to complete, 
+### we can just re-run the second job to check the status
+
+name: Trigger Private Janus build
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  trigger-workflow:
+    runs-on: ubuntu-latest
+
+    outputs:
+      workflow_run_id: ${{ steps.fetch.outputs.result }}
+
+    steps:
+    - name: record start time
+      run: echo START_TIME=$(date "+%Y-%m-%dT%H:%M:%S") >> $GITHUB_ENV
+
+    - name: Trigger workflow
+      uses: actions/github-script@v6
+      with:
+        github-token: ${{ secrets.PAT_ACTIONS_JANUS }}
+        script: |
+          await github.rest.actions.createWorkflowDispatch({
+            owner: 'guardian',
+            repo: 'janus',
+            workflow_id: 'build.yml',
+            ref: 'main',
+            inputs: {
+              "janus-app-branch": "${{ github.ref }}"
+            }
+          })
+
+    - run: sleep 5 # wait for event to register within github system
+
+    - name: fetch workflow id
+      id: fetch
+      uses: actions/github-script@v6
+      with:
+        github-token: ${{ secrets.PAT_ACTIONS_JANUS }}
+        result-encoding: string
+        script: |
+          const result = await github.rest.actions.listWorkflowRuns({
+            owner: 'guardian',
+            repo: 'janus',
+            workflow_id: 'build.yml',
+            event: 'workflow_dispatch',
+            created: '>=${{ env.START_TIME }}'
+          });
+          console.log(result);
+          return result["data"]["workflow_runs"][0].id
+
+    - run: sleep 240 # wait 4 minutes for job to run
```

Comment:
How was this duration chosen?

Reply:
Oh wow, the post setup action for caching sbt dependencies has massively increased. I selected the 4 minutes based on previous runs in teamcity being about 2 and a half minutes, and the github actions runs being about a minute longer due to caching the dependencies, but this run seems like it took 4 mins to cache the dependencies, I wonder if we've added a much larger set of packages?

On the plus side, this is why I broke the workflow into 2 jobs, so that if the time is exceeded by the build, we just need to re-run the failed job that checks for deployment

Reply:
@akash1810 I've just re-run the failed job, and it completed successfully.
I think we should wait and see if subsequent builds are also much longer, and if so, bump this wait time